server/src/test/java/org/elasticsearch/test/hamcrest/ElasticsearchGeoAssertions.java

```diff
@@ -219,25 +229,50 @@ public static void assertEquals(Shape s1, Shape s2) {
         }
     }
 
-    private static Geometry unwrap(Shape shape) {
+    @Deprecated
+    private static Geometry unwrapJTS(Object shape) {
         assertThat(shape, instanceOf(JtsGeometry.class));
         return ((JtsGeometry)shape).getGeom();
     }
 
-    public static void assertMultiPolygon(Shape shape) {
-        assert(unwrap(shape) instanceof MultiPolygon): "expected MultiPolygon but found " + unwrap(shape).getClass().getName();
+    public static void assertMultiPolygon(Object shape, boolean useJTS) {
+        if (useJTS) {
+            assert(unwrapJTS(shape) instanceof MultiPolygon)
+                : "expected MultiPolygon but found " + unwrapJTS(shape).getClass().getName();
+        } else {
+            assert(shape instanceof org.apache.lucene.geo.Polygon[])
+                : "expected Polygon[] but found " + shape.getClass().getName();
+        }
     }
 
-    public static void assertPolygon(Shape shape) {
-        assert(unwrap(shape) instanceof Polygon): "expected Polygon but found " + unwrap(shape).getClass().getName();
+    public static void assertPolygon(Object shape, boolean useJTS) {
+        if (useJTS) {
+            assert (unwrapJTS(shape) instanceof Polygon) : "expected Polygon but found "
+                + unwrapJTS(shape).getClass().getName();
+        } else {
+            assert(shape instanceof org.apache.lucene.geo.Polygon)
+                : "expected Polygon but found " + shape.getClass().getName();
+        }
     }
 
-    public static void assertLineString(Shape shape) {
-        assert(unwrap(shape) instanceof LineString): "expected LineString but found " + unwrap(shape).getClass().getName();
+    public static void assertLineString(Object shape, boolean useJTS) {
+        if (useJTS) {
+            assert (unwrapJTS(shape) instanceof LineString) : "expected LineString but found "
+                + unwrapJTS(shape).getClass().getName();
+        } else {
+            assert(shape instanceof org.apache.lucene.geo.Line)
+                : "expected Line but found " + shape.getClass().getName();
+        }
     }
 
-    public static void assertMultiLineString(Shape shape) {
-        assert(unwrap(shape) instanceof MultiLineString): "expected MultiLineString but found " + unwrap(shape).getClass().getName();
+    public static void assertMultiLineString(Object shape, boolean useJTS) {
+        if (useJTS) {
+            assert (unwrapJTS(shape) instanceof MultiLineString) : "expected MultiLineString but found "
+                + unwrapJTS(shape).getClass().getName();
+        } else {
+            assert(shape instanceof org.apache.lucene.geo.Line[])
```

Comment:
let's use JUNIT's assertTrue instead so that this is also tested if assertions are disabled